### DISCUSSION TITLE
Is this a problem with latest update from pub.dev?

### DISCUSSION THREAD

**Discussion by jbarasa** [hm21/pro_image_editor#230] [Q&A] [answered]
../../.pub-cache/hosted/pub.dev/pro_image_editor-5.4.0/lib/modules/emoji_editor/emoji_editor.dart:115:7: Error: No named parameter with the name 'swapCategoryAndBottomBar'.
      swapCategoryAndBottomBar:
      ^^^^^^^^^^^^^^^^^^^^^^^^
../../.pub-cache/hosted/pub.dev/emoji_picker_flutter-3.1.0/lib/src/config.dart:10:9: Context: Found this candidate, but the arguments don't match.
  const Config({
        ^^^^^^

**Comment by hm21** [comment]
It seems there is an issue with the `emoji_picker_flutter` package in your app. Have you already tried running `flutter clean` followed by `flutter pub get`? Additionally, are you directly using `emoji_picker_flutter` as a dependency in your app, or is it only included indirectly through the image editor?

**Reply by jbarasa** [reply]
it's only included through the image editor. i cleaned but still same issue 

```
../../.pub-cache/hosted/pub.dev/pro_image_editor-5.4.0/lib/modules/emoji_editor/emoji_editor.dart:115:7: Error: No named parameter with the name 'swapCategoryAndBottomBar'.
      swapCategoryAndBottomBar:
      ^^^^^^^^^^^^^^^^^^^^^^^^
../../.pub-cache/hosted/pub.dev/emoji_picker_flutter-3.1.0/lib/src/config.dart:10:9: Context: Found this candidate, but the arguments don't match.
  const Config({
        ^^^^^^
../../.pub-cache/hosted/pub.dev/pro_image_editor-5.4.0/lib/modules/emoji_editor/widgets/emoji_picker_view.dart:286:31: Error: The getter 'swapCategoryAndBottomBar' isn't defined for the class 'Config'.
 - 'Config' is from 'package:emoji_picker_flutter/src/config.dart' ('../../.pub-cache/hosted/pub.dev/emoji_picker_flutter-3.1.0/lib/src/config.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'swapCategoryAndBottomBar'.
                widget.config.swapCategoryAndBottomBar
                              ^^^^^^^^^^^^^^^^^^^^^^^^
../../.pub-cache/hosted/pub.dev/pro_image_editor-5.4.0/lib/modules/emoji_editor/widgets/emoji_picker_view.dart:294:31: Error: The getter 'swapCategoryAndBottomBar' isn't defined for the class 'Config'.
 - 'Config' is from 'package:emoji_picker_flutter/src/config.dart' ('../../.pub-cache/hosted/pub.dev/emoji_picker_flutter-3.1.0/lib/src/config.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'swapCategoryAndBottomBar'.
                widget.config.swapCategoryAndBottomBar
                              ^^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot_program failed: Exception
```

**Reply by hm21** [reply]
It seems that you are using `emoji_picker_flutter` version `3.1.0`, while the `pro_image_editor` package depends on version `3.0.0`. This creates a compatibility conflict in your code. You should review all your dependencies to identify the source of the conflict, which should help resolve the issue.

If you are confident that there is no version conflict, you can also try running the command `flutter pub cache clean` to clear your cache.

**Reply by hm21** [reply, marked as answer]
@jbarasa I’ve just released version `5.4.1`, which now includes the latest version of `emoji_picker_flutter` so it should be fixed after you update.

**Reply by jbarasa** [reply]
Thanks for the update! I appreciate the quick fix with version 5.4.1. It has worked very well.

To prevent future issues with dependency conflicts, it might be helpful for you to consider locking the versions of your dependencies in your package. This way, if a dependency gets updated, it won’t inadvertently affect your package or the users relying on it.

For instance, specifying exact versions in the pubspec.yaml or using dependency_overrides can help maintain stability.

Thanks again for your support!
